src/zenkit/core/vec3f.py

```diff
@@ -0,0 +1,254 @@
+import math
+from ctypes import Structure, c_float
+from typing import ClassVar, Any
+
+class Vec3f(Structure):
+    """
+    A class to represent a 3D vector using floats, supporting various vector operations.
+    
+    Attributes:
+        x (float): The x-component of the vector.
+        y (float): The y-component of the vector.
+        z (float): The z-component of the vector.
+    """
+
+    _fields_: ClassVar[tuple[str, Any]] = [
+        ("x", c_float),
+        ("y", c_float),
+        ("z", c_float),
+    ]
+
+    def __repr__(self) -> str:
+        """
+        Return a string representation of the vector.
+
+        Returns:
+            str: A string in the format 'Vec3f(x, y, z)'.
+        """
+        return f"Vec3f(x={self.x}, y={self.y}, z={self.z})"
+    
+    def __neg__(self):
+        """
+        Negate the vector (reverse the direction of the vector).
+
+        Returns:
+            Vec3f: A new Vec3f instance with negated components.
+        """
+        return Vec3f(-self.x, -self.y, -self.z)
+    
+    def __eq__(self, other):
+        """
+        Check if two vectors are equal component-wise.
+
+        Args:
+            other (Vec3f): The vector to compare with this vector.
+
+        Returns:
+            bool: True if the vectors are equal, False otherwise.
+
+        Raises:
+            TypeError: If the comparison is not between two Vec3f objects.
+        """
+        if isinstance(other, Vec3f):
+            return self.x == other.x and self.y == other.y and self.z == other.z
+        raise TypeError("Comparison is only supported between two Vec3f objects")
```

Comment:
`__eq__` should return `False` instead of throwing when comparing against an object of another type.